# Implementation of Zero Shot Learning with text data, NLP

# **Hugging Face API is used to perform classification**
 



The main dependency, Transformers are installed.
 Hugging Face allows us to leverage the pre-trained model to perform classification.

In [1]:
!pip install transformers

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.8/6.8 MB 42.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.6/7.6 MB 104.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 199.8/199.8 KB 26.5 MB/s eta 0:00:00


# Importing  pipeline .

In [2]:
from transformers import pipeline

# Initializing  pipeline

The imported pipeline is initialized to a variable. The result is stored in a  variable called variable_pipeline.




*   Here we are using the facebook/bart-large-mnli model. 
*   **BERT - Bidirectional Encoder Representations** from Transformers




BERT is used because it tends to generate contextualized embeddings. It looks at the statement’s context and generates a meaningful vector representation for a given word.


*   BERT Large is a huge model using 24 encoder layers.
*   BERT large is used for this categorization of text to their domain



In [3]:
variable_pipeline = pipeline ("zero-shot-classification", model = "facebook/bart-large-mnli")

In [5]:
input_sequence = "We are traveling to Himalayas"
label_candidate = ['travel', 'cooking', 'entertainment', 'dancing', 'technology']
variable_pipeline(input_sequence, label_candidate)

{'sequence': 'We are traveling to Himalayas',
 'labels': ['travel', 'entertainment', 'technology', 'dancing', 'cooking'],
 'scores': [0.9822729825973511,
  0.007975134067237377,
  0.005399617366492748,
  0.002184581942856312,
  0.0021677089389413595]}

From the results, the model has correctly classified the sentence as  travel category with 96% accuracy.

In [6]:
input_sequence = "In this digital world , Machine Learning is in boom"
candidate_labels = ['travel', 'technology', 'algorithms' 'cooking', 'dancing', 'exploration']
classifier_pipeline(input_sequence, label_candidate, multi_label=True)

{'sequence': 'In this digital world , Machine Learning is in boom',
 'labels': ['technology', 'entertainment', 'dancing', 'travel', 'cooking'],
 'scores': [0.99150550365448,
  0.030818002298474312,
  0.007000857498496771,
  0.0038627595640718937,
  0.0002963569713756442]}

Here  the model has correctly classified the sentence as  category of technology with 99% accuracy.i.e,with accuracy score 0.99150550365448.